rfcs/047-nested-columns.md

```diff
@@ -0,0 +1,189 @@
+This enhancement requires layout change, existing component (chevron and column details area) behavior change and a new right side panel (See [Column details access and side panels RFC](./046-column-details-access-and-side-panels.md)), but we think it is the opportune timing and an investment to make Amundsen and Nested Columns more extensible. 
+
+
+
+
+## Alternatives
+
+We have explored several directions and decided against them because…
+
+### Chevron to expand nesting, + icon to open column details
+
+![Chevron to expand nesting, + icon to open column details](../assets/047/alt-1.png)
+
+* The chevron feels appropriate for expand/collapsing hierarchy in the list, but it’s not easy to add another CTA that will open column details in the same area (near column names)
+* Two types of expanding actions in the same list can easily take up a lot of vertical space, making it hard to scan and compare multiple columns
+
+
+
+### Table metadata in search results page, column metadata in table page
+
+![Table metadata preview in search results page](../assets/047/alt-2-1.png)
+
+* We thought about removing the table metadata section from the table details page to make more room and show column info, but it’s important to see the table metadata together with the list of columns.
+![Full width columns list](../assets/047/alt-2-2.png)
+
+![Column metadata in right side panel](../assets/047/alt-2-3.png)
+
+* A dedicated panel will help properly show all column metadata, but it didn’t have to be this wide.
+
+
+
+### Nested object without named fields (e.g. Array and Map) variations
+We debated a lot on the optimal way to highlight and distinguish the hierarchy of nested object without named fields, and the concepts we considered below were inferior to the direction we chose in terms of the clarity and visual design.
+
+![All caps short rows](../assets/047/alt-array-1.png)
+
+![All caps short rows with brackets at start and end](../assets/047/alt-array-2.png)
+
+![Short rows with brackets at start and end](../assets/047/alt-array-3.png)
+
+![Array type icon](../assets/047/alt-array-4.png)
+
+![Dark border around the whole group](../assets/047/alt-array-5.png)
+
+![Grey frame around nested objects](../assets/047/alt-array-6.png)
+
+![Grey vertical bar on the left side of nested objects](../assets/047/alt-array-7.png)
+
+![All caps labeling after column name](../assets/047/alt-array-8.png)
+
+
+
+## Prior art
+
+This project comes from the user feedback and shortcomings of our current POC for supporting nested columns.
+
+## Unresolved questions
+
+Decide whether we need to load a collapsed view as default in the edge case that there are prohibitively many columns to render, and what the threshold should be.
+
+## Future possibilities
+
+We could add a breadcrumb at the top of right side to help users see the whole path with clickable shortcuts:
```

Comment:
I really like this and would love to see resource/hierarchy with breadcrumbs more incorporated into Amundsen (already mentioned in #46 with database > table > column relationships)